File: typescript/clase-18/final/main.ts
import Anthropic from "@anthropic-ai/sdk";
import Fastify from "fastify";
import { z } from "zod";

export {};

const ChatRequest = z.object({ message: z.string().min(1) });
const server = Fastify({ logger: true });

server.get("/health", async () => ({ status: "ok" }));

server.post("/chat", async (request, reply) => {
  const apiKey = process.env.ANTHROPIC_API_KEY;
  if (!apiKey) return reply.code(500).send({ error: "ANTHROPIC_API_KEY no configurada." });

  const payload = ChatRequest.parse(request.body);
  const client = new Anthropic({ apiKey });
  const response = await client.messages.create({
    model: "claude-3-5-sonnet-latest",
    max_tokens: 500,
    messages: [{ role: "user", content: payload.message }],
  });
  const answer = response.content.filter((block) => block.type === "text").map((block) => block.text).join("");
  return { answer };
});

await server.listen({ port: Number(process.env.PORT ?? 3000), host: "0.0.0.0" });

File: python/clase-18/final/main.py
"""API REST con FastAPI lista para desplegar en Railway."""

import os
from anthropic import Anthropic
from fastapi import FastAPI, HTTPException
from pydantic import BaseModel

MODEL = "claude-3-5-sonnet-latest"
app = FastAPI(title="Curso Claude API")


class ChatRequest(BaseModel):
    message: str


class ChatResponse(BaseModel):
    answer: str


@app.get("/health")
def health() -> dict[str, str]:
    return {"status": "ok"}


@app.post("/chat", response_model=ChatResponse)
def chat(payload: ChatRequest) -> ChatResponse:
    api_key = os.getenv("ANTHROPIC_API_KEY")
    if not api_key:
        raise HTTPException(status_code=500, detail="ANTHROPIC_API_KEY no configurada.")

    client = Anthropic(api_key=api_key)
    response = client.messages.create(
        model=MODEL,
        max_tokens=500,
        messages=[{"role": "user", "content": payload.message}],
    )
    answer = "".join(block.text for block in response.content if block.type == "text")
    return ChatResponse(answer=answer)
